contrib/windows/exe_path_wrapper.c

```diff
@@ -0,0 +1,73 @@
+#include <windows.h>
+#include <tchar.h>
+#include <stdio.h>
+#include <shlwapi.h>
+#define ENVVAR_MAXLEN 32760
+
+/* PATH_ENTRIES is our simulated RPATH, usually of the form 'TEXT("../path1"), TEXT("../path2"), ...' */
+#ifndef PATH_ENTRIES
+#define PATH_ENTRIES  TEXT("")
+#endif
+
+LPWSTR pathEntries[] = {
+    PATH_ENTRIES
+};
+
+/* JULIA_EXE_PATH is the relative path to julia.exe */
+#ifndef JULIA_EXE_PATH
+#define JULIA_EXE_PATH "../libexec/julia.exe"
+#endif
+
+int wmain(int argc, wchar_t *argv[], wchar_t *envp[]) {
+    // Determine absolute path to true julia.exe sitting in `libexec/`
+    WCHAR currFileDir[MAX_PATH];
+    WCHAR juliaPath[MAX_PATH];
+    GetModuleFileName(NULL, currFileDir, MAX_PATH);
+    PathRemoveFileSpec(currFileDir);
+    PathCombine(juliaPath, currFileDir, TEXT(JULIA_EXE_PATH));
+
+    // On windows, we simulate RPATH by pushing onto PATH
+    LPWSTR pathVal = (LPWSTR) malloc(ENVVAR_MAXLEN*sizeof(WCHAR));
+    DWORD dwRet = GetEnvironmentVariable(TEXT("PATH"), pathVal, ENVVAR_MAXLEN);
+    DWORD numPathEntries = sizeof(pathEntries)/sizeof(pathEntries[0]);
+    if (dwRet == 0) {
+        // If we cannot get PATH, then our job is easy!
+        pathVal[0] = '\0';
+    } else {
+        // Otherwise, we append, if we have enough space to:
+        DWORD currFileDirLen = wcslen(currFileDir);
+        DWORD totalPathLen = dwRet + 1 + currFileDirLen;
+        for (int env_idx=0; env_idx<numPathEntries; ++env_idx) {
```

Comment:
IIRC, some (Windows) compilers can't deal with this syntax, so we often avoid it, though not entirely (and it would be so nice to use)
```suggestion
        int env_idx;
        for (env_idx = 0; env_idx < numPathEntries; ++env_idx) {
```